src/hypofuzz/dashboard/dashboard.py

```diff
@@ -559,6 +547,37 @@ def send_nowait_from_anywhere(msg: ListenerEventT) -> None:
         )
         TESTS[fuzz_target.nodeid] = test
 
+        async def update_websockets() -> None:
+            # TODO: make this more granular, so we send incremental batches
+            # of reports as they're loaded, etc.
+            for websocket in websockets.copy():
+                await websocket.initial({test.nodeid: test})
```

Comment:
Sending to everything seems super inefficient; can't we look up the specific websocket for this test and send it once?  Plus maybe to the overview page, I guess, but O(2) vs O(n) is a big deal on the inside of a size-n loop.

Reply:
If we change `websockets` from a set to a dict, with keys string-nodeid or None, I think we'd just drop this helper function and it'd be less code overall even if we make two calls to cover the overview + tests page.

Reply:
I'd prefer to keep the filtering logic in the `Websocket` class, to support more complicated filtering (e.g. maybe a future "test directory" page, for tests only in a subdir). Maybe we could build an efficient `nodeid: list[websocket]` dispatch, but I'm tempted to defer this